### PR TITLE
Hunting queries windows event batch4

### PR DESCRIPTION
**Pull request by yaelrbergman**
Required items, please complete

Change(s):
Add Windows Event Data to the following Hunting Queries:

Hunting Queries/SecurityEvent/ADAccountLockouts.yaml
Hunting Queries/SecurityEvent/Least_Common_Parent_Child_Process.yaml
Hunting Queries/SecurityEvent/Least_Common_Process_Command_Lines.yaml
Hunting Queries/SecurityEvent/Least_Common_Process_With_Depth.yaml
Hunting Queries/SecurityEvent/masquerading_files.yaml
Hunting Queries/SecurityEvent/MSRPRN_Printer_Bug_Exploitation.yaml
Hunting Queries/SecurityEvent/NishangReverseTCPShellBase64.yaml
Reason for Change(s):
Part of adding Windows Event data source project

Version Updated:
No
Testing Completed:
Yes
Checked that the validations are passing and have addressed any issues that are present:
Yes

### CODE REVIEW

File: Hunting Queries/SecurityEvent/Least_Common_Parent_Child_Process.yaml [hunk 1]
```diff
@@ -29,4 +34,23 @@ query: |
...
   | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), TimesSeen = count(), HostCount = dcount(Computer), Hosts = makeset(Computer), UserCount = dcount(SubjectUserName), Users = makeset(SubjectUserName) by ParentChildPair
   | where TimesSeen < Sensitivity
+  | extend timestamp = StartTimeUtc),
+  (WindowsEvent
+  | where TimeGenerated between(lookback..endtime)
+  | where EventID == 4688 
+  | extend ParentProcessName = tostring(EventData.ParentProcessName)
+  | where isnotnull(ParentProcessName)
+  | extend NewProcessName = tostring(EventData.NewProcessName)
+  | extend ProcArray = split(NewProcessName, '\\'), ParentProcArray = split(ParentProcessName, '\\')
```
Comment (oshezaf) [inline Hunting Queries/SecurityEvent/Least_Common_Parent_Child_Process.yaml:44]: Well, the original is so dump that I think you should fix it. What they do is just split(...)[-1] to get the filename.... and in 7 lines.

File: Hunting Queries/SecurityEvent/Least_Common_Process_Command_Lines.yaml [hunk 2]
```diff
@@ -27,3 +31,20 @@ query: |
...
   | extend timestamp = StartTimeUtc
+  ),
+  (WindowsEvent
+  | where TimeGenerated between(lookback..endtime)
+  | where EventID == 4688 and EventData has 'conhost.exe'
+  | extend NewProcessName = tostring(EventData.NewProcessName)
+  | where NewProcessName !endswith 'conhost.exe'
+  | extend ProcArray = split(NewProcessName, '\\')
+  // ProcArrayLength is Folder Depth
+  | extend ProcArrayLength = arraylength(ProcArray)
```
Comment (oshezaf) [inline Hunting Queries/SecurityEvent/Least_Common_Process_Command_Lines.yaml:42]: Same

File: Hunting Queries/SecurityEvent/MSRPRN_Printer_Bug_Exploitation.yaml [hunk 3]
```diff
@@ -18,8 +21,23 @@ query: |
...
+ union isfuzzy=true 
+ ( SecurityEvent
  // | where Computer in (DCServersList)
  // | where IpAddress  in (UnconstrainedServersIPList)
  | where EventID == 5145 and ShareName == "\\\\*\\IPC$" and RelativeTargetName == "spoolss"
- | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by Computer, SubjectUserName, IpAddress, ShareName, RelativeTargetName, Type, SubjectUserSid
+ | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by Computer, SubjectUserName, IpAddress, ShareName, RelativeTargetName, Type, SubjectUserSid),
+ (WindowsEvent
+ // | where Computer in (DCServersList)
+ // | where EventData has_any (UnconstrainedServersIPList)
```
Comment (oshezaf) [inline Hunting Queries/SecurityEvent/MSRPRN_Printer_Bug_Exploitation.yaml:32]: has_any_ipv4

File: Hunting Queries/SecurityEvent/MSRPRN_Printer_Bug_Exploitation.yaml [hunk 4]
```diff
@@ -18,8 +21,23 @@ query: |
...
+ // | extend IpAddress = tostring(EventData.IpAddress)
```
Comment (oshezaf) [inline Hunting Queries/SecurityEvent/MSRPRN_Printer_Bug_Exploitation.yaml:33]: Move those lines (33, 34) to after the EventID filter line.

**Review by oshezaf — CHANGES_REQUESTED** [review]

**Comment by vmanojreddy** [comment]
@aprakash13  Please review analytic rules. Thanks.

**Comment by vmanojreddy** [comment]
@oshezaf Please check and approve the changes, thanks.

**Comment by vmanojreddy** [comment]
@oshezaf Please check and approve the changes, thanks!!!

**Comment by v-spadarthi** [comment]
@oshezaf : Please check and approve the changes, thanks!!

**Comment by v-mchatla** [comment]
Hi @oshezaf
Can we close this and reopen whenever required as you have labelled it as differed.
Thanks

**Comment by oshezaf** [comment]
Done

**Pull request closed without merging**